core/App/screens/ProofRequest.tsx

```diff
@@ -194,11 +196,21 @@ const ProofRequest: React.FC<ProofRequestProps> = ({ navigation, route }) => {
       ...retrievedCredentials?.requestedPredicates,
     }
 
-    // TODO:(jl) Need to test with partial match? Maybe `.some` would work?
-    return (
-      typeof retrievedCredentials !== 'undefined' &&
-      (credName ? fields[credName]?.length > 0 : Object.values(fields).every((c) => c.length > 0))
-    )
+    if (credName) {
+      let credFound = false
+      Object.keys(fields).forEach((proofKey) => {
+        const credNamesInAttrs = fields[proofKey].map((attr) =>
+          parseCredDefFromId(attr.credentialInfo?.credentialDefinitionId, attr.credentialInfo?.schemaId)
+        )
+        if (credNamesInAttrs.includes(startCase(credName))) {
+          credFound = true
+          return
+        }
+      })
+      return credFound
+    }
+
+    return typeof retrievedCredentials !== 'undefined' && Object.values(fields).every((c) => c.length > 0)
```

Comment:
I know you didn't write this line, but should we also be checking for `null`, not just undefined?

Reply:
we could use it as a falsy value and that way it would catch undefined, null, and the empty string